internal/tools/tool/ads.go

```diff
@@ -0,0 +1,64 @@
+package tool
+
+import (
+	"github.com/microsoft/go-sqlcmd/internal/io/file"
+	"os"
+	"path/filepath"
+)
+
+type Ads struct {
+	Base
+}
+
+func (t *Ads) Init() {
+	t.Base.SetName("ads")
+	userProfile := os.Getenv("USERPROFILE")
+	programFiles := os.Getenv("ProgramFiles")
+
+	// Search in this order
+	//   User Insiders Install
+	//   System Insiders Install
+	//   User non-Insiders install
+	//   System non-Insiders install
+	searchLocations := []string{
+		filepath.Join(userProfile, "AppData\\Local\\Programs\\Azure Data Studio - Insiders\\azuredatastudio-insiders.exe"),
+		filepath.Join(programFiles, "Azure Data Studio - Insiders\\azuredatastudio-insiders.exe"),
+		filepath.Join(userProfile, "AppData\\Local\\Programs\\Azure Data Studio\\azuredatastudio.exe"),
+		filepath.Join(programFiles, "Azure Data Studio\\azuredatastudio.exe"),
+	}
+
+	t.Base.SetExeName("azuredatastudio")
```

Comment:
>xeName("azuredatastudio") [](http://example.com/codeflow?start=12&length=25)

in the past, I had problems getting azuredatastudio.exe to properly parse command line parameters and we had to invoke the wrapper `.cmd` in the `bin` folder instead.
Has that situation changed?

Reply:
I can say it is working with this code, but the current surface area of go-sqlcmd doesn't let much complexity come down to this level at the moment.